src/tlo/methods/healthsystem.py

```diff
@@ -2842,6 +2857,29 @@ def apply(self, population):
             self.module.bed_days.availability = self._parameters['beds_availability']
 
 
+class DynamicRescalingHRCapabilities(RegularEvent, PopulationScopeEventMixin):
+    """ This event exists to scale the daily capabilities assumed at fixed time intervals"""
+    def __init__(self, module):
+        super().__init__(module, frequency=DateOffset(years=1))
+
+    def apply(self, population):
+
+        # The following assumes that self.module._daily_capabilities is initialised once at start of simulation,
+        # which I believe is correct
+
+        # Rescale daily capabilities by specified amount
+        self.module._daily_capabilities *= self.module.parameters['dynamic_HR_scaling_factor']
+
+        # Rescale daily capabilities by population size, if this option is included
+        if self.module.parameters['scale_HR_by_popsize']:
+            demog = self.sim.modules['Demography']
+            if self.sim.date.year>2010:
+                # Either
+                self.module._daily_capabilities *= population.props.is_alive.sum()/demog.popsize_by_year[self.sim.date.year - 1]
+                # Or, if ensuring DemographyLoggingEvent at start of the year takes place *before* this rescaling
+                # self.module._daily_capabilities *= demog.popsize_by_year[self.sim.date.year]/demog.popsize_by_year[self.sim.date.year - 1]
+
```

Comment:
Or, to keep it all internal to that function, let that class measure population size itself: i.e.,
```python
df = self.sim.population.props
this_year_pop_size = df.is_alive.sum()
ratio_pop_growth = this_year_pop_size / last_year_pop_size()
ratio_for_hr_scaling = ratio_pop_growth + self.module.parameters['dynamic_HR_scaling_factor']
# do scaling....

last_year_pop_size = this_year_pop_size   # save for next year

```